### PR TITLE
Add OrderPomElements for ordering POM elements per the recommended order

### PR DESCRIPTION
**Pull request by pway99**
Note this PR contains a disabled test illustrating a new-line formatting issue.

### CODE REVIEW

File: rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import org.openrewrite.Recipe;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.xml.AutoFormat;
+import org.openrewrite.xml.XmlVisitor;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+public class OrderPomElements extends Recipe {
```
Comment (tkvangorder) [inline rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java:28]: Can you please tag this with the proper sonar ID? Also you should have some docs that reference where the order was defined.

Comment (tkvangorder) [reply rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java:28]: oh, I see you reference that in the description.

Comment (tkvangorder) [reply rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java:28]: but we do need to add a tag, like we do for other sonar-based rules.

File: rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-3423");
+    }
+
+    @Override
+    protected MavenVisitor getVisitor() {
+        return new MavenVisitor() {
```
Comment (tkvangorder) [inline rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java:77]: No need to have a nested visitor here. MavenVisitor IS an XmlVisitor. 

Just override those methods in your MavenVisitor and get rid of OrderVisitor.

Comment (pway99) [reply rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java:77]: nice suggestion thanks @tkvangorder

**Review by tkvangorder — APPROVED** [review]
LGTM

**Pull request merged**